lumberdev/sum-it-up
# Pratikmdhr/feat share summary

## WHAT IT DOES:

_Description of the pull request, what changed_

- Added a button to share the url for the summary result page
- Two query parameters used:
`original`: stores the original content
`result`: store the response object
- The button will be disabled (on desktop and hidden on mobile) which the content is streaming
- The state will also be encoded to the url once the streaming is done, so the user can directly copy the URL
- ~Added a separate route (/result) to display the summary result from the shared url.. (Added this separately because decoding the url in the homepage would cause the Input fields to render before the summary result page gets displayed because `displayResult` is initially set to be false.. Thought about adding a loading page, but the loading page would also get displayed for the main Input page (homepage).. any suggestions?)~
- Separated the client side code inside page.tsx into a new file called `clientPage.tsx`


## HOW TO TEST:

_Description of how to test if applicable_
The **Copy Link**  button in the results page should copy the url with the query parameters with encoded state. The url would link to the Result page which should display the summary result. 

The url should update with encoded state once the streaming is done

### PREVIEW:

_Attach any relevant videos or images:_
Added the text of the button to be `Copy Link`.. could also change it to `Share Link` or just `Share` (But i think that would imply sharing it on Social?)
<img width="160" alt="image" src="https://user-images.githubusercontent.com/52301822/219710972-fbf36614-a972-4396-a2b5-70534de74a66.png">


---

### TASK:

### WHAT'S LEFT:

_Delete if not necessary:_

- [ ] - Add analytics to the copy button
- [ ] - Confirm button text (or improve UX while button is pressed) and change accordingly
- [ ] - May be add share button to social media?
- [ ] - Open to removing the separate route for /result page.. and use the original homepage to decode the url
